workflow-templates/im-deploy-az-app-manually.yml

```diff
@@ -44,6 +44,8 @@ on:
           - demo
           - uat
           - prod
+          - stage-secondary
+          - prod-secondary
```

Comment:
I recall asking @danielle-casella-adams  this question a few years ago on whether or not the secondary failover regions are represented as environments.  The response I received is they should not, rather as a separate identifier.

Has that now changed?

Reply:
With that said, I don't see it a problem including that second identifier as part of the environment.

Reply:
I'm not sure how you could deploy to a secondary region without a pseudo environment of some kind.  Unless everything gets deployed at the same time (web-app, networking, etc) I think you need a logical way to separate that.

Reply:
I agree with Danielle-of-a-few-years-ago 😉 that the secondary regions are not "environments". But, as [I commented earlier](https://github.com/im-practices/.github/pull/194#issuecomment-1400760897), we can still have a consolidated list of options with `dev`, `stage`, etc. plus `stage-secondary` and `prod-secondary` like Josh has done here if we just rename that input to something other than "environment".

Reply:
> I'm not sure how you could deploy to a secondary region without a pseudo environment of some kind. Unless everything gets deployed at the same time (web-app, networking, etc) I think you need a logical way to separate that.

Agreed. I'm suggesting we call it "target" or "destination" or something like that in these deployment workflows.

Reply:
Like Mike said they aren't separate environments they are a subset of the resources in an environment.  If you are deploying though you do need a way to distinguish between your regular stage resources and your stage secondary like Jason pointed out.  It makes it a little weird if we change the input name though because we are setting the GitHub Actions environment based on that input.

If I was setting up a secondary region for my azure resources I wouldn't want to create another GitHub Actions environment that has duplicate values (ARM_CLIENT_ID, ARM_CLIENT_SECRET, etc) and that the SP credential rotation workflow doesn't know how to sync to.  It seems like an additional input is warranted, one for the environment where it pulls secrets from and one for the target or destination that indicates whether you want your primary or secondary resources to go.

Reply:
I just made my way through the rest of this file and see that it is setting a GITHUB_SECRETS_ENV based on the input.  I think in that case Mike makes a good point and changing the name of the input is the way to go, something like target or destination.

Reply:
I do argue that not providing a solution is way worse than just providing a way for teams to easily adopted.  Adding another variable will make it harder to track.  I like the idea of renaming `environment` field to a combo.  `environment-and-target`.  Then it is correct in both senses, but technically correct.

Reply:
Another option is to continue to use the single input, but break out the two meanings into ENV vars.  I have been doing this with my workflows using a simple action I created.

This example is more complex and serving other purposes, but it shows how it can be used:
1. The target input represents more than just the environment: https://github.com/im-enrollment/ifp-enroll-connector-infrastructure/blob/main/.github/workflows/im-deploy-tf-manual-apply.yml#L16

2. GitHub action that parses out the additional values represented in the single input: https://github.com/im-enrollment/ifp-enroll-connector-infrastructure/blob/main/.github/workflows/im-deploy-tf-manual-apply.yml#L116

3. The rest of the workflow is able to use the environment variable separate from the location variable:
https://github.com/im-enrollment/ifp-enroll-connector-infrastructure/blob/main/.github/workflows/im-deploy-tf-manual-apply.yml#L66

That way I am able to still able to target the GitHub Environment while loading up my resource configuration for the secondary target.

The GitHub Action I am using to parse out those values:
https://github.com/im-enrollment/parse-scopes-action